src/app/common/utils/resolvers/pages-wallet-details.resolver.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import {
+  Router, Resolve,
+  RouterStateSnapshot,
+  ActivatedRouteSnapshot
+} from '@angular/router';
+import { firstValueFrom } from 'rxjs';
+import { DomainsWalletsGateway } from 'src/app/domains/wallets/domains.wallets.gateway';
+import { IWalletApiResponse } from 'src/app/domains/wallets/domains.wallets.types';
+import { WalletsManagementItem } from 'src/app/pages/wallets/management/pages-wallets-wallets-management-item.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PagesWalletDetailsResolver implements Resolve<WalletsManagementItem> {
+
+  constructor(private readonly domainsWalletsGateway: DomainsWalletsGateway) {}
+
+  async resolve(route: ActivatedRouteSnapshot): Promise<WalletsManagementItem> {
+
+    const walletsObs = await this.domainsWalletsGateway.getWallets();
```

Comment:
This is wrong, you just need `const wallets = await firstValueFrom(this.domainsWalletsGateway.getWallets());`